### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by rsandor**
Please ensure that this will work as a direct drop-in replacement for our copied `loadenv` code everywhere. Some quick notes:
- Removed git hub environment variable since they are being set by NPM anyway
- Added ability to restore environment

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+var dotenv = require('dotenv');
+var eson = require('eson');
+var exists = require('101/exists');
+var clone = require('clone');
+
+/**
+ * Application root directory.
+ * @type {string}
+ */
+var applicationRoot = require('app-root-path').toString();
```
Comment (anandkumarpatel) [inline index.js:23]: should this be configurable? hmm not sure, I am kinda down forcing people to have `$ROOT/configs` dir

**Comment by anandkumarpatel** [comment]
one discussion other then that LGTM

File: index.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+/**
+ * Restores process.env to its original state.
+ */
+readDotEnvConfigs.restore = function () {
+  process.env = originalEnv;
+  hasLoadedEnvironment = false;
+};
+
+module.exports = readDotEnvConfigs;
```
Comment (cflynn07) [inline index.js]: I really like the exports defined near the top of the module, make use of hoisting

File: index.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+/**
+ * Loads environment varibles defined in `config/.env*`.
+ * @param {string} [debugName] Name to use when outputting resulting environment
+ *  via debug. Defaults to 'loadenv'.
+ * @param {boolean} [ignoreNodeEnv] Flag to ignore loading environment based
+ *  on `NODE_ENV` variable.
+ */
+function readDotEnvConfigs(debugName, ignoreNodeEnv) {
+  // Skip if environment has already been loaded.
+  if (hasLoadedEnvironment === true) {
```
Comment (cflynn07) [inline index.js]: might as well skip the `=== true`

**Comment by cflynn07** [comment]
LGTM!

File: index.js (hunk 1 above)
Comment (rsandor) [inline index.js:23]: Per our discussion, we will enforce the use of `config/` with the possibility of adding an option in the future.

**Pull request merged**